File: solutions/julia/proverb/1/proverb.jl
function recite(pieces)
    length(pieces)==0 && return ""
    lines = ["For want of a "*pieces[i-1]*" the "*pieces[i]*" was lost." for i in 2:length(pieces)]
    push!(lines, "And all for the want of a "*pieces[1]*".")
    join(lines, '\n')
end

File: solutions/dart/proverb/1/lib/proverb.dart
class Proverb {
  String recite(List<String> pieces){
    String res = "";
    if(pieces.isEmpty)
      return res;
    for(var i=0; i<pieces.length-1; i++){
      res += "For want of a ${pieces[i]} the ${pieces[i+1]} was lost.\n";
    }
    res += "And all for the want of a ${pieces[0]}.";
    return res;
  }
}
